### PR TITLE
Implement chat presence notification

### PR DESCRIPTION
**Pull request by Gilbert189**
This PR adds the `chatPresenceNotify` add-on, which posts a message when a user enters/leaves the chat. It's based on [this gist](https://gist.github.com/Gilbert189/ab04381364c49078a3f12d9017338fd7) I made a while back.

Currently it's has some problems working on Firefox due to some permission error that I currently cannot resolve, so testing is done on Chrome at my other laptop.

### Things to do before merging
- [x] Find out another way to get `$socket` (or `socket`)

### Things we could do before merging
- [ ] Fix the permission error (for Firefox)
- [ ] Fix the intermittent to failure to call of `onUnload()` before closing tab (is this browser related?)

### CODE REVIEW

**Comment by imadeanaccount1** [comment]
This looks like a really cool contribution! Thank you and I can't wait until it's in a mergeable state! 🤩

**Comment by Gilbert189** [comment]
I can confirm that the permission error is only on Firefox, since I've tried it on Chrome and it doesn't have the permission error.
However, now there's another problem; the `window` doesn't actually have the `$nuxt` key (on wasteof3 `socket` is `undefined`)

**Comment by Gilbert189** [comment]
I figured out that injecting the notify script to `<head>` would work instead of executing it directly on `addon.js`. A weird hack, sure; but I see other addons doing this exact thing so I guess I'm not the only one that thought of it.

And now, it works... on Chrome at least. Firefox still had that permission problem. But since Firefox compatibility isn't on high priority yet, I'll split it off to a different section.

**Comment by imadeanaccount1** [comment]
@all-contributors please add @Gilbert189 for code, ideas

**Review by imadeanaccount1 — CHANGES_REQUESTED** [review]
Are all of these new permissions really necessary?
"notifications",
        "tabs",
        "idle",
        "webNavigation",
        "webRequest",
        "webRequestBlocking",
        "unlimitedStorage",
        "storage",
        "contextMenus",
        "clipboardWrite",
        "cookies",
        "downloads",
        "<all_urls>"

**Comment by imadeanaccount1** [comment]
https://github.com/all-contributors please add @Gilbert189 for code, ideas

**Comment by imadeanaccount1** [comment]
@all-contributors please add @Gilbert189 for code, ideas

**Comment by imadeanaccount1** [comment]
@Gilbert189 
Can't test right now because wasteof's down but I'll try to work on merging sometime when the site gets back up 😕

I may also wait until PR #22 is merged (it adds new changes to the structure of addon.json files). See more info in the CONTRIBUTING.md file of the relevant branch.

File: manifest_firefox.json
Comment (imadeanaccount1) [inline manifest_firefox.json:1]: Too many permissions

**Comment by imadeanaccount1** [comment]
@Gilbert189 I just confirmed that the firefox bug will be fixed in a PR I will create tonight. It is unrelated to your code, though.

**Review by imadeanaccount1 — APPROVED** [review]

File: manifest_firefox.json
Comment (Gilbert189) [reply manifest_firefox.json:1]: Yeah, that was my "solution" to the permission error for Firefox

**Pull request merged**
